tfrecord/reader.py

```diff
@@ -90,9 +90,11 @@ def read_records(start_offset=None, end_offset=None):
 
 
 def tfrecord_loader(data_path: str,
-                    index_path: str,
+                    index_path: typing.Union[str, None],
```

Comment:
Looks like we accept None as a value for the input, so indicating that it's `str` type is deceptive at best.